sdk/servicebus/azure-servicebus/azure/servicebus/common/utils.py

```diff
@@ -56,9 +57,11 @@ def parse_conn_str(conn_str):
             shared_access_key = value
         elif key.lower() == 'entitypath':
             entity_path = value
+        elif key.lower() == 'transporttype':
+            transport_type = value
     if not all([endpoint, shared_access_key_name, shared_access_key]):
         raise ValueError("Invalid connection string")
-    return endpoint, shared_access_key_name, shared_access_key, entity_path
+    return endpoint, shared_access_key_name, shared_access_key, entity_path, transport_type
```

Comment:
Is transport type part of the connection string grabbed from Azure portal? or you need to reconstruct the connection string by appending the `TransportType=amqp`?

As mentioned above, TransportType can a separate Enum parameter passed into from_connection_string/constructors.

Reply:
All I know is that in C#, you can add TransportType=AmqpOverWebsocket to the connection string to make it use websockets. I believe the enum in C# is documented here:

https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.servicebus.transporttype?view=azure-dotnet

Interestingly the values have different values than in the uamqp library.

I am going to amend my changes to `parse_conn_str` to convert the value portion of "transport_type" directly to the enum to maintain consistency and not use the string value anywhere else for this parameter.

Reply:
I've made the changes here. I think `from_connection_string` should only expect to receive the transport type as part of the string itself. For example, you don't currently expect that users would do something like:
`ServiceBusClient.from_connection_string(conn_str, shared_access_key_name="test")`

On the other hand, I've made the change so that they could pass the enum value for transport_type to the `ServiceBusClient` constructor, just as they would for the other parts of what would be in a connection string.